### DISCUSSION TITLE
How to load script coming from db on Grapesjs

### DISCUSSION THREAD

**Discussion by kmlcnclk** [GrapesJS/grapesjs#5409] [Q&A]
I am saving some scripts to db using editor.getJs() function. I want to load this js when grapesjs loading. I can load components but js is not loading.

**Comment by padcom** [comment]
As luck would have it, I am currently working on a wizard component that uses quite a bit of script...

I think you're going about it the wrong way. From what I've read in the docs the `editor.toJs()` is the final output from whatever components have been added to the page. So, when GrapesJS loads the project it will add that JavaScript automatically, per instance of the component that adds the script.

Check out this part of the docs: https://grapesjs.com/docs/modules/Components-js.html

**Reply by kmlcnclk** [reply]
There is no problem with that. In my project, I have to save the design to the DB. Also, I have to load desgin from Db to my design page. I can get the Components which ı save to DB but I can not get the Scripts.

**Reply by padcom** [reply]
Scripts are not serializable. You can't save scripts (it is mentioned in the docs that I linked). 

The script is only coming from component definition in GrapesJS. You can get the entire page, including scripts, by calling `editor.toHtml()` or if you choose to opt out of rendering scripts with html (`isInHtml` option when calling `grapesjs.init({..})`) then you can get the relevant scripts using `editor.getJs()`. For completeness, you can get the relevant styles by calling `editor.getCss()`.

Alternatively, when you compose the final page you can always add a `<script>` tag yourself and attach any external scripts that way. It's also in the docs I mentioned. Maybe that's what you're looking for. If you stored the output of `editor.getJs()` then once you put together the final page just load it from the database, wrap it into a `<script>...</script>` block and attach it at the end of your document. That should do the trick.

**Reply by artf** [reply]
@kmlcnclk Using HTML/CSS/JS output to edit your projects is the wrong way to use GrapesJS https://grapesjs.com/docs/modules/Storage.html#project-data
